power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/meta_data.hpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_META_DATA_HPP
+#define POWER_GRID_MODEL_CPP_META_DATA_HPP
+
+#include "power_grid_model_c/meta_data.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class MetaData {
+  public:
+    static Idx _n_datasets() {
```

Comment:
please remove all prefixes `_` here. The public interface should not have prefix `_`. If there's a name conflict (unlikely), please resolve in a different way.